src/Orleans.Runtime/Lifecycle/SiloLifecycle.cs

```diff
@@ -1,12 +1,28 @@
 ﻿
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 
 namespace Orleans.Runtime
 {
     internal class SiloLifecycle : LifecycleObservable, ISiloLifecycle
     {
-        public SiloLifecycle(ILoggerFactory loggerFactory) : base(loggerFactory)
+        private SiloLifecyclePerfReporter perfReporter;
+        public SiloLifecycle(ILoggerFactory loggerFactory, SiloLifecyclePerfReporter reporter) : base(loggerFactory)
         {
+            this.perfReporter = reporter;
+        }
+
+        public override async Task OnStart(CancellationToken ct)
```

Comment:
@jason-bragg let me know if you are ok with this, marking OnStart and OnStop as `virtual `on LifeCycleObservable. I think it will be useful for people who want to use LifecycleObservable implementation but also add some custom logic

Reply:
I'm fine with the calls being virtual, but the reporter mechanism seems clumsy.  If let's talk about the design in person.

Reply:
I'd call it minimalist :). The requirement of this feature wasn't clear to me, so In my mind, I just want a internal feature, which provide a place to cache all the reports for tasks run in silo lifecycle rings, and flush those reports when silo finish onStart or finish onStop. But I agree it has space to improve